client.go

```diff
@@ -754,11 +738,15 @@ func (c *Client) CreateSession(ctx context.Context, cfg *uasc.SessionConfig) (*S
 		// Ensure we have a valid identity token that the server will accept before trying to activate a session
 		if c.cfg.session.UserIdentityToken == nil {
 			opt := AuthAnonymous()
-			opt(c.cfg)
+			if err := opt(c.cfg); err != nil {
```

Comment:
revert

Reply:
FYI: opt returning an error now, so here in that case that was me not the linter, but will do

Reply:
I would suggest adding your suggested TODO. For the first call it might fail eventually once this TODO here gets fixed 
```
if !ok {
	// todo(fs): should we Fatal here?
	log.Printf("non-anonymous authentication already configured, ignoring")
	return nil
}
```
In case you decide to return the error there you must remember to handle the error in the call we talk about.